scripts/core/editor3/store/index.js

```diff
@@ -65,12 +65,17 @@ export default function createEditorStore(props, isReact = false) {
         suggestingMode: false,
         invisibles: false,
         svc: props.svc,
+        abbreviations: {},
     }, applyMiddleware(thunk));
 
 
     // after we have the dictionary, force update the editor to highlight typos
     dict.finally(() => store.dispatch(forceUpdate()));
 
+    spellcheck.getAbbreviationsDict().then((abbreviations) => {
```

Comment:
We shouldn't mix asynchronous and synchronous code(`createEditorStore` is synchronous). It'd be better to first load `abbreviations` from server and only then call `createEditorStore` with `abbreviations` as an argument. It would be in the initial state and the extra dispatch wouldn't be required.

Reply:
"Doesn't work properly on pasting: "App*" gets pasted and only after clicking the word, it's replaced with "Application"."
@tomaskikutis  I checked and this works well.


"Cursor is at the wrong position after typing "*" and getting the word autocompleted."
Currently the cursor remains at the same position. 
@karelpetrak Should we move the cursor after the last character added? 

"FYI @karelpetrak, abbreviations don't work in comments, tables, and inline metadata(image title). Fixing it in comments should not take much, but implementing it in tables or metadata labels would take around a week each."

@tomaskikutis I checked and on comments the abbreviations works well.

Reply:
Yeah, I've seen that the same thing is used for dictionaries which I would refactor as well.

> I think serialising all those loads will just slow the editor3 initialisation and will not add any benefit.

The benefit is avoiding the state when editor3 is enabled, but abbreviations not working because the request for abbreviations hasn't completed yet. That's rare for sure, but I like correctness ;)

To speed the initialization we could have all editor3 instances initialize locked, and when you focus the editor, the requests for dictionaries/abbreviations would start and the editor would unlock when they finish. This would be both correct and fast.

Anyway, this is minor stuff, feel free to leave this as is.

Reply:
I think the initialisation speed is much important than an improbable problem that can appear because of this.

Reply:
> "Doesn't work properly on pasting: "App*" gets pasted and only after clicking the word, it's replaced with "Application"."

> I checked and this works well.

![abbr-paste](https://user-images.githubusercontent.com/2076953/45364022-c5775000-b5e1-11e8-9a00-e8a2e417ed9f.gif)

Reply:
> I checked and on comments the abbreviations works well.

I was talking about the contents of the comment, not the highlighted text.

Reply:
@karelpetrak I think on content of the comment it is not needed the abbreviations.
"Doesn't work properly on pasting: "App*" gets pasted and only after clicking the word, it's replaced with "Application"."
As I said on pasting "App*" it works well. On pasting a larger text that contains abbreviations will need more work. @karelpetrak Should we do this?

Reply:
@tomaskikutis @ioanpocol Sorry for belated answers. Here they are:
>Doesn't work properly on pasting: "App*" gets pasted and only after clicking the word, it's replaced with "Application".
The abbreviations feature is a typing assistant, I would not expect the pasted text with "App*" to be replaced without user interaction.

>abbreviations don't work in comments, tables, and inline metadata(image title). Fixing it in comments should not take much, but implementing it in tables or metadata labels would take around a week each.
This is OK for now. We may add it for those fields later.

>"Cursor is at the wrong position after typing "*" and getting the word autocompleted."
Currently the cursor remains at the same position. Should we move the cursor after the last character added?
Yes, I would expect the cursor to jump at the end of the auto-completed word so that I can continue typing.

Reply:
>Yes, I would expect the cursor to jump at the end of the auto-completed word so that I can continue typing.

This is already done